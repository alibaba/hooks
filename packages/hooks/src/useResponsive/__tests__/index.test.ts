import { renderHook, act } from '@testing-library/react-hooks';
import { useResponsive } from '../';

describe('useResponsive', () => {
  function changeWidth(width: number) {
    act(() => {
      (global as any).innerWidth = width;
      (global as any).dispatchEvent(new Event('resize'));
    });
  }
  changeWidth(1024);

  const hook = renderHook(() => useResponsive());

  it('should response to window width changes', () => {
    expect(hook.result.current).toMatchSnapshot();
    changeWidth(300);
    expect(hook.result.current).toMatchSnapshot();
    changeWidth(700);
    expect(hook.result.current).toMatchSnapshot();
    changeWidth(800);
    expect(hook.result.current).toMatchSnapshot();
    changeWidth(1000);
    expect(hook.result.current).toMatchSnapshot();
    changeWidth(1200);
    expect(hook.result.current).toMatchSnapshot();
  });
});
