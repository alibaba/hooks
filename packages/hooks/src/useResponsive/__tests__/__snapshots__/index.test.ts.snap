// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useResponsive should response to window width changes 1`] = `
{
  "lg": true,
  "md": true,
  "sm": true,
  "xl": false,
  "xs": true,
}
`;

exports[`useResponsive should response to window width changes 2`] = `
{
  "lg": false,
  "md": false,
  "sm": false,
  "xl": false,
  "xs": true,
}
`;

exports[`useResponsive should response to window width changes 3`] = `
{
  "lg": false,
  "md": false,
  "sm": true,
  "xl": false,
  "xs": true,
}
`;

exports[`useResponsive should response to window width changes 4`] = `
{
  "lg": false,
  "md": true,
  "sm": true,
  "xl": false,
  "xs": true,
}
`;

exports[`useResponsive should response to window width changes 5`] = `
{
  "lg": true,
  "md": true,
  "sm": true,
  "xl": false,
  "xs": true,
}
`;

exports[`useResponsive should response to window width changes 6`] = `
{
  "lg": true,
  "md": true,
  "sm": true,
  "xl": true,
  "xs": true,
}
`;
